### PR TITLE
Prevent observe before server is finished starting

### PR DESCRIPTION
**Pull request by Kyep**
## What Does This PR Do
Prevents players (including admins) from using "Observe" before the server is finished starting.
Until the server is ready for observers, a "Observe (starting up...)" will be displayed instead of the normal blue "Observe" button.

NOTE: This _doesn't_ stop you using Observe several minutes before roundstart. It just stops you using it for the first 5-20 seconds or so after the server first comes online, while the server is still initializing, loading the map, etc.

## Why It's Good For The Game
1) Various misleading runtimes and other errors can happen if you leap into the round before the server has even finished initializing subsystems and things.
2) This is eventually going to be required in order to support map rotation.

## Images
TZUcUcszmh
(yes, its slightly bigger than the options list presented, this is to account for "View the Crew Manifest" appearing for latejoiners)

## Changelog
:cl: Kyep
add: Server no longer lets you Observe before its even finished loading the game.
/:cl:

### CODE REVIEW

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -103,7 +106,7 @@
 
 	output += "</center>"
 
-	var/datum/browser/popup = new(src, "playersetup", "<div align='center'>New Player Options</div>", 220, 290)
+	var/datum/browser/popup = new(src, "playersetup", "<div align='center'>New Player Options</div>", 240, 330)
```
Comment (AffectedArc07) [inline code/modules/mob/new_player/new_player.dm:109]: Does this finally fix the weirdness where youre in lobby midround and it doesnt show properly.
image

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -184,6 +187,10 @@
 			to_chat(usr, "<span class='warning'>You must consent to the terms of service before you can join!</span>")
 			return 0
 
+		if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
+			to_chat(usr, "<span class='warning'>You must wait for the server to finish starting before you can join!</span>")
+			return 0
```
Comment (AffectedArc07) [inline code/modules/mob/new_player/new_player.dm:192]: Can you make these TRUE/FALSE please?

File: code/modules/mob/new_player/new_player.dm (hunk 1 above)
Comment (Kyep) [reply code/modules/mob/new_player/new_player.dm:109]: Assuming "the weirdness" you're talking about is the fact it has a scroll bar, yes.

File: code/modules/mob/new_player/new_player.dm [hunk 3]
```diff
@@ -77,7 +77,10 @@
 		else	output += "<p><a href='byond://?src=[UID()];skip_antag=2'>Global Antag Candidacy</A>"
 		output += "<br /><small>You are <b>[client.skip_antag ? "ineligible" : "eligible"]</b> for all antag roles.</small></p>"
 
-	output += "<p><a href='byond://?src=[UID()];observe=1'>Observe</A></p>"
+	if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
+		output += "<p>Observe (starting up...)</p>"
```
Comment (AffectedArc07) [inline code/modules/mob/new_player/new_player.dm:81]:
```suggestion
		output += "<p>Observe (Please wait...)</p>"
```

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**